<launch>
  <!-- load perception -->
  <!-- this launch file was moved inside this package in order to change laser settings -->
  <include file="$(find pr2_object_manipulation_launch)/launch/laser+stereo-perception.launch">
    <arg name="stereo" value="$(arg stereo)"/>
  </include>

  <!-- load planning -->
  <include file="$(find pr2_arm_navigation_planning)/launch/ompl_planning.launch"/>

  <!-- load move_arm -->
  <include file="$(find pr2_utils)/launch/move_right_arm_no_head_following.launch"/>
  <include file="$(find pr2_utils)/launch/move_left_arm_no_head_following.launch"/>

  <!-- load planning_environment -->
  <include file="$(find pr2_arm_navigation_actions)/launch/environment_server.launch"/>

  <!-- load controllers -->
  <include file="$(find pr2_arm_navigation_filtering)/launch/both_arms_filters.launch"/>
  <param name="trajectory_filter/object_padd" value="0.04"/>

  <!-- load ik -->
  <include file="$(find pr2_arm_navigation_kinematics)/launch/right_arm_collision_free_ik.launch"/>
  <include file="$(find pr2_arm_navigation_kinematics)/launch/left_arm_collision_free_ik.launch"/>
  <param name="pr2_right_arm_kinematics/object_padd" value="0.04"/>
  <param name="pr2_left_arm_kinematics/object_padd" value="0.04"/>

  <!--trigger mode 3 is on, 4 is off, 5 is alternate projector-->
  <!--
  <node name="change_narrow_trigger" pkg="dynamic_reconfigure" type="dynparam" 
    args="set /camera_synchronizer_node narrow_stereo_trig_mode 5" />
  -->

  <arg name="stereo" default="true"/>
  <node if="$(arg stereo)"
	name="projector_on" pkg="dynamic_reconfigure" type="dynparam"
	args="set camera_synchronizer_node projector_mode 3"/>
  <node if="$(arg stereo)"
	name="narrow_alternate" pkg="dynamic_reconfigure" type="dynparam"
	args="set camera_synchronizer_node narrow_stereo_trig_mode 5"/>
  <node if="$(arg stereo)"
	name="wide_without" pkg="dynamic_reconfigure" type="dynparam"
	args="set camera_synchronizer_node wide_stereo_trig_mode 4"/>
  <node unless="$(arg stereo)"
	name="projector_off" pkg="dynamic_reconfigure" type="dynparam"
	args="set camera_synchronizer_node projector_mode 1"/>
  
  <!--node name="change_narrow_stereo_params" pkg="dynamic_reconfigure" type="dynparam" 
    args="set /narrow_stereo_textured/narrow_stereo_textured_proc min_disparity 30" /-->

  <!-- change controller gains -->
  <!--
  <node name="change_controller_gains" pkg="pr2_object_manipulation_launch" type="change_joint_controller_gains.py" />
  -->

  <!-- interpolated IK -->
  <node name="interpolated_ik_node_right" pkg="interpolated_ik_motion_planner" 
    type="interpolated_ik_motion_planner.py" args="r" respawn="false" machine="c2"/>
  <node name="interpolated_ik_node_left" pkg="interpolated_ik_motion_planner" 
    type="interpolated_ik_motion_planner.py" args="l" respawn="false" machine="c2"/>

  <!-- joint trajectory normalization -->
  <include file="$(find trajectory_filter_server)/launch/trajectory_unnormalize.launch"/>

</launch>
